blackjax/mcmc/barker.py

```diff
@@ -81,44 +81,57 @@ def build_kernel():
     """
 
     def _compute_acceptance_probability(
-        state: BarkerState,
-        proposal: BarkerState,
+        state: BarkerState, proposal: BarkerState, C_t: Array, C_t_inv: Array
     ) -> float:
         """Compute the acceptance probability of the Barker's proposal kernel."""
 
-        def ratio_proposal_nd(y, x, log_y, log_x):
-            num = -_log1pexp(-log_y * (x - y))
-            den = -_log1pexp(-log_x * (y - x))
+        x_flat, _ = ravel_pytree(state.position)
+        y_flat, _ = ravel_pytree(proposal.position)
+        log_x_flat, _ = ravel_pytree(state.logdensity_grad)
+        log_y_flat, _ = ravel_pytree(proposal.logdensity_grad)
 
-            return jnp.sum(num - den)
+        z = C_t_inv.dot(y_flat - x_flat)
+        c_x = log_x_flat.dot(C_t)
+        c_y = log_y_flat.dot(C_t)
+
+        num = _log1pexp(-z * c_x)
+        denom = _log1pexp(z * c_y)
+
+        ratio_proposal = jnp.sum(num - denom)
 
-        ratios_proposals = tree_map(
-            ratio_proposal_nd,
-            proposal.position,
-            state.position,
-            proposal.logdensity_grad,
-            state.logdensity_grad,
-        )
-        ratio_proposal = sum(tree_leaves(ratios_proposals))
         return proposal.logdensity - state.logdensity + ratio_proposal
 
     def kernel(
-        rng_key: PRNGKey, state: BarkerState, logdensity_fn: Callable, step_size: float
+        rng_key: PRNGKey,
+        state: BarkerState,
+        logdensity_fn: Callable,
+        step_size: float,
+        inverse_mass_matrix: Array,
```

Comment:
Breaking change, this needs to default to previous behaviour.

Reply:
made the `inverse_mass_matrix` an optional argument in altest commit